### ISSUE TITLE
The input when using F.grid_sample for 3D interpolation should be zyx instead of xyz?

### ISSUE THREAD

**Issue by YZsZY** [oppo-us-research/NARUTO#1]
Hello author, thanks for publishing this excellent work so quickly! But I want to point out an error in the code as below:
https://github.com/oppo-us-research/NARUTO/blob/04988b4943d252072eaf191379fbaef438c206f6/src/slam/coslam/model/scene_rep.py#L62
I think when using F.grid_sample to interpolate the feature of a voxel grid, the input order of the 3d points should be ZYX instead of XYZ, and the codes should be rectified as follows:
```python
uncert = torch.nn.functional.grid_sample(self.uncert_grid[None, None, ...], torch.flip(grid, dims=[-1]), align_corners=False)
```

The following code demonstrates that F.grid_sample should use ZYX instead of XYZ:
```python
import torch
import torch.nn.functional as F

reso = 64
neural_volume = torch.rand(1, 4, reso, reso, reso)
sample_points_origin = torch.randint(0, reso, (100, 3))
sample_points = sample_points_origin / (reso - 1) * 2 - 1  # [0,127]->[0,1]->[0,2]->[-1,1]

sampled_feature = F.grid_sample(neural_volume, torch.flip(sample_points.view(1, 1, 1, -1, 3), dims=[-1]),
                                mode='bilinear', padding_mode='zeros',
                                align_corners=True).squeeze().permute(1, 0)  # [N,16]

pts_coords = ((sample_points + 1) / 2 * (reso - 1)).round().long()
# sample_points_origin  # ((sample_points + 1) / 2 * (reso - 1)).long()
idx_feature = neural_volume[0, :, pts_coords[:, 0], pts_coords[:, 1], pts_coords[:, 2]].permute(1, 0)

print("grid_sample: ", sampled_feature[10])
print("idx_sample:  ", idx_feature[10])

```

**Comment by Huangying-Zhan** [comment]
Hi @YZsZY , Thank you for your interest in our work and for your comments regarding the coordinate system.

We recognize that our current coordinate system differs from the one typically used in PyTorch implementations, where the XYZ coordinate system is oriented Right-Down-Forward, and the order is (DHW). In our implementation, we maintain the same coordinate system ordering as the scene to facilitate the conversion between voxel space and metric space.
Attached is an illustration that highlights these differences.
We acknowledge that it might appear a bit upside down from a PyTorch perspective! 😄
![image](https://github.com/oppo-us-research/NARUTO/assets/13748721/57324803-0d14-4230-9fe7-05e7d27fce9e)

**Comment by Huangying-Zhan** [comment]
Hi @YZsZY , Thank you for your interest in our work and for your comments regarding the coordinate system.

We recognize that our current coordinate system differs from the one typically used in PyTorch implementations, where the XYZ coordinate system is oriented Right-Down-Forward, and the order is (DHW). In our implementation, we maintain the same coordinate system ordering as the scene to facilitate the conversion between voxel space and metric space.
Attached is an illustration that highlights these differences.
We acknowledge that it might appear a bit upside down from a PyTorch perspective! 😄
![image](https://github.com/oppo-us-research/NARUTO/assets/13748721/6b93270c-a47e-4d7e-8df1-b81c5fb24732)

**Comment by YZsZY** [comment]
@Huangying-Zhan It's not easy to notice this, thanks for the reminder~
